quickwit/quickwit-janitor/src/actors/garbage_collector.rs

```diff
@@ -666,5 +691,10 @@ mod tests {
         assert_eq!(counters.num_failed_storage_resolution, 0);
         assert_eq!(counters.num_failed_gc_run_on_index, 0);
         assert_eq!(counters.num_failed_splits, 2);
+        assert!(!universe
+            .quit()
+            .await
+            .into_iter()
+            .any(|s| matches!(s, ActorExitStatus::Panicked)));
```

Comment:
You seem to be copying this snippet in many places...

```
        assert!(!universe
            .quit()
            .await
            .into_iter()
            .any(|s| matches!(s, ActorExitStatus::Panicked)));
```

Do we want it as a method in Universe? Do we want it on Drop?

Reply:
I really wish I could add it to Drop, but there is no support for async drop in rust as far as I understand. I experimented with it a bit more by adding a check to Drop that ensure that universe.quit() was really called. That identified a whole bunch of tests where it was missing, so I added it there, but now some ingest tests are hanging from time to time. I want to figure out what's going on before I can push the changes.

Reply:
Good point. The `assert_quit` solution is good.